### PR TITLE
[#1] Migrate knotx-docker build to gradle

### PR DESCRIPTION
**Pull request by dragarthPl**
Migrate knotx-docker build to gradle

## Description

knotx-docker was migrated to gradle. Now you can do just:
$> ./gradlew build

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

---
I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,11 @@ You need to have installed **docker** on your machine.
 
 To build the docker images locally, just launch:
 
-`mvn clean package`
+`$> mvn clean package`
```
Comment (malaskowski) [inline README.md]: We don't plan to keep maven here, but it can be removed in the separate PR.

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    id("com.bmuschko.docker-remote-api")
+    id("java")
+}
+
+dependencies {
+    subprojects.forEach { "dist"(project(":${it.name}")) }
+}
+
+allprojects {
+    group = "com.project"
```
Comment (malaskowski) [inline build.gradle.kts]: That's not necessary.

File: buildSrc/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+repositories {
+    mavenLocal()
+    mavenCentral()
+    gradlePluginPortal()
+}
+dependencies {
+    implementation("com.bmuschko:gradle-docker-plugin:6.1.3")
+    implementation("io.knotx:knotx-gradle-plugins:0.1.2")
```
Comment (malaskowski) [inline buildSrc/build.gradle.kts]: Let's use the latest `0.1.4` version.

File: gradle.properties [hunk 4]
```diff
@@ -0,0 +1,8 @@
+version=2.1.1-SNAPSHOT
+# Knot.x Docker image version (https://hub.docker.com/r/knotx/knotx)
+knotx.version=2.1.0
```
Comment (malaskowski) [inline gradle.properties]: I think that should alway be the same as `version`.

File: gradle.properties [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+knotx.conf=knotx
+docker.image.name=knotx/knotx-docker
```
Comment (malaskowski) [inline gradle.properties]: What for is that variable?

File: gradle/docker.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    targetImageId(if (File(dockerBaseImageRef).exists()) File(dockerBaseImageRef).readText() else "")
+    onError {
+        if (!this.message!!.contains("No such image"))
+            throw this
+    }
+}
+
+tasks.register<DockerRemoveImage>("removeBaseAlpineImage") {
+    group = "docker"
```
Comment (malaskowski) [inline gradle/docker.gradle.kts:65]: Looks like a good candidate for `DockerRemoveImage.remove` helper (both tasks have similar logic).

File: setting.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+ */
+
+pluginManagement {
+    repositories {
+        maven { url = uri("https://plugins.gradle.org/m2/") }
+        mavenLocal()
+    }
+}
+
+rootProject.name = "knotx-docekr"
```
Comment (malaskowski) [inline setting.gradle.kts]:
```suggestion
rootProject.name = "knotx-docker"
```

File: build.gradle.kts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+tasks.named("build") {
+    dependsOn("build-stack", "build-docker")
+}
+
+tasks.register("build-docker") {
+    group = "docker"
+    dependsOn("prepareDocker")
+}
+
+tasks.register("build-stack") {
```
Comment (malaskowski) [inline build.gradle.kts]: `build-stack` -> `fetch-stack`?

**Review by malaskowski — CHANGES_REQUESTED** [review]
Hi @dragarth-7 
Thank you for contribution!

File: README.md (hunk 1 above)
Comment (dragarthPl) [reply README.md]: Done

File: build.gradle.kts (hunk 2 above)
Comment (dragarthPl) [reply build.gradle.kts]: Done

File: buildSrc/build.gradle.kts (hunk 3 above)
Comment (dragarthPl) [reply buildSrc/build.gradle.kts]: Done

File: gradle.properties (hunk 4 above)
Comment (dragarthPl) [reply gradle.properties]: Done

File: gradle.properties (hunk 5 above)
Comment (dragarthPl) [reply gradle.properties]: That variable is redundant. I deleted it.

File: setting.gradle.kts (hunk 7 above)
Comment (dragarthPl) [reply setting.gradle.kts]: Done

File: build.gradle.kts (hunk 8 above)
Comment (dragarthPl) [reply build.gradle.kts]: Done

File: gradle/docker.gradle.kts (hunk 6 above)
Comment (dragarthPl) [reply gradle/docker.gradle.kts:65]: @Skejven Check my solution for this tip and let me know if that was the case.

**Comment by malaskowski** [comment]
Thank you @dragarth-7 - this looks really good.
Can I ask you for squashing all commits into the single one, just to keep the history clear?

I will change the target of this PR to `feature/release-automation` as moving to gradle is part of the https://github.com/Knotx/knotx-aggregator/issues/19 .

**Comment by dragarthPl** [comment]
Done, please check is there everthing ok.

**Review by malaskowski — APPROVED** [review]
Yup, that looks good, thank you @dragarth-7 for this PR.
